crates/external_services/src/http_client/client.rs

```diff
@@ -17,30 +17,48 @@ pub fn create_client(
     proxy_config: &Proxy,
     client_certificate: Option<masking::Secret<String>>,
     client_certificate_key: Option<masking::Secret<String>>,
+    ca_certificate: Option<masking::Secret<String>>,
 ) -> CustomResult<reqwest::Client, HttpClientError> {
-    match (client_certificate, client_certificate_key) {
-        (Some(encoded_certificate), Some(encoded_certificate_key)) => {
-            let client_builder = get_client_builder(proxy_config)?;
-
-            let identity = create_identity_from_certificate_and_key(
-                encoded_certificate.clone(),
-                encoded_certificate_key,
-            )?;
-            let certificate_list = create_certificate(encoded_certificate)?;
-            let client_builder = certificate_list
-                .into_iter()
-                .fold(client_builder, |client_builder, certificate| {
-                    client_builder.add_root_certificate(certificate)
-                });
-            client_builder
-                .identity(identity)
-                .use_rustls_tls()
-                .build()
-                .change_context(HttpClientError::ClientConstructionFailed)
-                .attach_printable("Failed to construct client with certificate and certificate key")
-        }
-        _ => get_base_client(proxy_config),
+    // Case 1: Mutual TLS with client certificate and key
```

Comment:
What would happen in case we need to use both client cert and key (for mTLS) with a custom CA certificate? (This is the case when we're calling our [`hyperswitch-encryption-service`](https://github.com/juspay/hyperswitch-encryption-service) for example.)

The current implementation (in this PR) does not support it at all and would always go with a mTLS setup, without considering the provided CA cert, and we don't have any logs warning us about this behavior.

Can we update the implementation to be not mutually exclusive, but be additive instead?

Reply:
thanks for noticing! updated the code